### PR TITLE
.Net: Deprecate OpenAI plugins

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
As OpenAI ChatGPT plugins have been deprecated, SK no longer needs to maintain code for working with them.

### Description
This PR annotates all OpenAI ChatGPT plugin-related code with the obsolete attribute to notify potential consumers of the API that the plugins will be deprecated.

Addresses the first step from the task: https://github.com/microsoft/semantic-kernel/issues/6881

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/samples/Concepts/Plugins/CreatePluginFromOpenAI_AzureKeyVault.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 namespace Plugins;
 
+[Obsolete("OpenAI plugins are deprecated and will be removed in a future version.")]
 public class CreatePluginFromOpenAI_AzureKeyVault(ITestOutputHelper output) : BaseTest(output)
```
Comment (westey-m) [inline dotnet/samples/Concepts/Plugins/CreatePluginFromOpenAI_AzureKeyVault.cs:15]: Do we also want to add some comment to the readme that links here, e.g. (Deprecated) in brackets after the link or something similar?

File: dotnet/samples/Concepts/Plugins/CreatePluginFromOpenApiSpec_Klarna.cs [hunk 2]
```diff
@@ -5,24 +5,24 @@
 
 namespace Plugins;
 
-public class OpenAIPlugins(ITestOutputHelper output) : BaseTest(output)
+public class CreatePluginFromOpenApiSpec_Klarna(ITestOutputHelper output) : BaseTest(output)
```
Comment (westey-m) [inline dotnet/samples/Concepts/Plugins/CreatePluginFromOpenApiSpec_Klarna.cs:8]: Remember to update the readme, with the new link to here.

**Review by westey-m — APPROVED** [review]

File: dotnet/samples/Concepts/Plugins/CreatePluginFromOpenAI_AzureKeyVault.cs (hunk 1 above)
Comment (SergeyMenshykh) [reply dotnet/samples/Concepts/Plugins/CreatePluginFromOpenAI_AzureKeyVault.cs:15]: Yep, make sense. Updated.

File: dotnet/samples/Concepts/Plugins/CreatePluginFromOpenApiSpec_Klarna.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/samples/Concepts/Plugins/CreatePluginFromOpenApiSpec_Klarna.cs:8]: Thanks, updated.

**Review by westey-m — APPROVED** [review]

**Pull request merged**